Source/controls/keymapper.cpp

```diff
@@ -0,0 +1,126 @@
+#include "keymapper.hpp"
+
+#include <cassert>
+#include <fmt/format.h>
+
+#include <SDL.h>
+
+#ifdef USE_SDL1
+#include "utils/sdl2_to_1_2_backports.h"
+#endif
+
+#include "../miniwin/miniwin.h"
+#include "utils/log.hpp"
+
+namespace devilution {
+
+Keymapper::Keymapper(SetConfigKeyFunction setKeyFunction, GetConfigKeyFunction getKeyFunction)
+    : setKeyFunction(setKeyFunction)
+    , getKeyFunction(getKeyFunction)
+{
+	keyIDToKeyName.reserve('Z' - 'A' + '9' - '0' + 12);

```

Comment:
I'm not sure I understand what is going on here. Are you creating a bitmask?

Reply:
Oh, I see reserve the 26 letters of the alphabet, 10 numbers, and 12 others?

Reply:
12 for the F-keys...

Reply:
Let's parenthesize this for clarity `('Z' - 'A') + ('9' - '0') + 12`

Reply:
It is also off-by-one:

```suggestion
	keyIDToKeyName.reserve(('Z' - 'A' + 1) + ('9' - '0' + 1) + 12);
```

Reply:
Well spotted. I'll also add a comment to make things more explicit.